### ISSUE TITLE
Silence pyopenms redeclaration warnings?

### ISSUE THREAD

**Issue by jpfeuffer** [OpenMS/OpenMS#4505]
Any ideas on how to silence those?
`warning: /Users/distiller/project/miniconda/conda-bld/openms_1580309845735/work/src/pyOpenMS/pxds/NASequence.pxd:103:8: 'ZIon' redeclared`

**Comment by hroest** [comment]
I dont think there is a good way, see https://stackoverflow.com/questions/50653853/duplicate-enum-member-names-in-cython-redeclaration-error - at least not until we have namespaces in Cython

**Comment by hroest** [comment]
I guess we would have to rename all the enum parameters?

**Comment by jpfeuffer** [comment]
I think so, too. That is why I was not so convinced anymore that it is worth it.

**Comment by hroest** [comment]
the question is still whether it is a problem or not -- it does not seem to be a problem and everything compiles but then why does Cython complain? It seems to be a Cython issue for now, so we just ignore it and wait for Cython to fix this?

**Comment by jpfeuffer** [comment]
Yes, to be honest for me it was just annoying when parsing logs while creating conda and pyopenms nightly wheels. I can also only judge from basic functionality and our unit tests, that it probably is indeed just a warning.

**Comment by timosachsenberg** [comment]
Maybe https://stackoverflow.com/questions/31001918/wrap-enum-class-for-cython could be a solution? Converting stuff to enum classes seems feasible.

**Comment by jpfeuffer** [comment]
Cython has full support for scoped enums now: https://github.com/cython/cython/pull/3640

**Comment by timosachsenberg** [comment]
Working on it

**Comment by dshradha** [comment]
is anyone working on it?

**Comment by jpfeuffer** [comment]
Not needed anymore with nanobind
